### PR TITLE
tags support

### PR DESCRIPTION
**Pull request by brahimABD98**
added tags support  but for one  tag at the time  it will crawl all the posts and look for all the tags and generate the paths accordingly with pagination 

- zod repo has a pr that will add array uniqueness checks [see](https://github.com/colinhacks/zod/pull/2672/commits)
- HorizontalCard is only used by the blog for now so idk if tag_url is even required
- added aliases for both components and layouts

### CODE REVIEW

**Comment by hmenorjr** [comment]
@brahimABD98 Hi, do you have a screenshot of this?

**Comment by brahimABD98** [comment]
@hmenorjr Oh this means you can filter posts using one tag  at a time (you can't filter using multiple tags at once ) but you can add multiple tags to each post  and they have to be unique 
screenshot :

File: src/pages/blog/tag/[tag]/[...page].astro [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                <p>
+                    There are no blog posts to show at the moment. Check back
+                    later!
+                </p>
+            </div>
+        ) : (
+            <ul>
+                {page.data.map((post) => (
+                    <>
+                        <HorizontalCard
```
Comment (manuelernestog) [inline src/pages/blog/tag/[tag]/[...page].astro:44]: As we are going to add support for tags, we should include the tags in all blog post cards. Let's add the `tags={post.data.tags}` here

File: src/pages/blog/tag/[tag]/[...page].astro [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            pageSize: 10,
+        });
+    });
+}
+
+const { page } = Astro.props;
+const params = Astro.params;
+---
+
+<BaseLayout title="Blog" sideBarActiveItemID="blog">
```
Comment (manuelernestog) [inline src/pages/blog/tag/[tag]/[...page].astro:27]: Here we can add a dynamic Title according to the tag selected. Something like `Blog - [tagname]`

File: src/pages/blog/tag/[tag]/[...page].astro [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    <div class="mb-5">
+        <div class="text-3xl w-full font-bold">Blog</div>
```
Comment (manuelernestog) [inline src/pages/blog/tag/[tag]/[...page].astro:29]: Let's add the tag name here too

**Review by manuelernestog — CHANGES_REQUESTED** [review]
Excellent work @brahimABD98 !

I leave you some small change requests. 

Also, you need to pull the code in the PR branch because I had to merge the master branch to fix some conflicts with new changes.

**Comment by brahimABD98** [comment]
@manuelernestog all good to go

**Review by manuelernestog — APPROVED** [review]

**Comment by hmenorjr** [comment]
@brahimABD98 Awesome, thanks. I was thinking of how to add a tag in the `HorizontalCard` since it's in a some sort of for-loop. I just got back to development for my website. So yeah, thanks. Like this template for SEO and simplicity. thanks again.

**Pull request merged**